cirq-google/cirq_google/experimental/noise_models/fidelity.py

```diff
@@ -0,0 +1,124 @@
+  def __init__(
+      self,
+      *,
+      t1: Optional = None,
+      decay_constant: Optional = None,
+      xeb_fidelity: Optional = None,
+      pauli_error: Optional = None,
+      p00: Optional = None,
+      p11: Optional = None
+  ) -> None:
+    """
+    Detailed comments about arguments here
+    """
+    if not any([t1, decay_constant, xeb_fidelity, pauli_error, p00, p00]):
+      raise ValueError('At least one metric must be specified')
+
+    self._t1 = t1
+    self._p = decay_constant
+    self._p00 = p00
+    self._p11 = p11
+    self._pauli_error = pauli_error
+    self._xeb = xeb_fidelity
+
+  @property
+  def decay_constant(self):
+    return self._p
+
+  @property
+  def p00(self):
+    return self._p00
+
+  @property
+  def p11(self):
+    return self._p11
+
+  @property
+  def pauli_error(self):
+    return self._pauli_error
+
+  @property
+  def t1(self):
+    return self._t1
+
+  def decay_constant_to_xeb_error(self, decay_constant: float, N: int = 4):
+    return (1 - decay_constant) * (1 - 1 / N)
+
+  def decay_constant_to_pauli_error(self, decay_constant: float, N: int = 2):
+    return (1 - decay_constant) * (1 - 1 / N / N)
+
+  def pauli_error_to_xeb_error(self, pauli_error: float, N: int = 4):
+    decay_constant = 1 - (pauli_error / (1 - 1 / N))
+    return decay_constant_to_xeb_error(decay_constant)
+
+  def pauli_error_to_decay_constant(self, pauli_error: float, N: int = 2):
+    return 1 - (pauli_error / (1 - 1 / N / N))
+
+  def pauli_error_from_t1(self, t: float, t1: float):
+    t2 = 2 * t1
+    return (1 - np.exp(-t / t2)) / 2 + (1 - np.exp(-t / t1)) / 4
+
+  def pauli_error_from_depolarization(self, t : float):
+    if self._t1 is not None:
+      return self._pauli_error - self.pauli_error_from_t1(t, self._t1)
+    else:
+      return self._pauli_error
+
+
+_sq_inds = np.arange(4)
+
+class NoiseModelFromFidelity(cirq.NoiseModel):
+  def __init__(
+      self,
+      fidelity: Fidelity
+  ) -> None:
+    if fidelity is not None:
+      self._fidelity = fidelity
+    else:
+      raise ValueError('A fidelity object must be specified')
+
+  def noisy_moment(self, moment: cirq.Moment, system_qubits: Sequence[cirq.Qid]) -> cirq.OP_TREE:
+    moments = []
+
+    ## TODO: write better code to get Sycamore duration for each gatee
+    duration = 25.0 # ns
+    if np.any([isinstance(op.gate, cirq.MeasurementGate) for op in moment.operations]):
+      duration = 4000.0
+
+    if validate_all_measurements(moment): ## add after measurement gate
+      if self._fidelity._p00 is not None: # apply p00 noise
+        gamma = self._fidelity._p00
+        p = 0.0
+        channel = cirq.GeneralizedAmplitudeDampingChannel(p = p, gamma = gamma)
+        moments.append(cirq.Moment(channel(q) for q in system_qubits))
+
+      if self._fidelity._p11 is not None: # apply p11 noise
+        gamma = self._fidelity._p11
+        p = 1.0
+        channel = cirq.GeneralizedAmplitudeDampingChannel(p = p, gamma = gamma)
```

Comment:
Possible source of error: having p00 and p11 as two separate channels represents a sequence of two separate decay / excitation events, which will behave slightly differently than the actual device.

To resolve this, the channels can be combined. This combined channel has `p` and `gamma` such that:
```
# An unexcited:excited ratio of p:(1-p) is the "equilibrium" state,
# where decay and excitation are balanced out.
p * p00 = (1 - p) * p11

# A system in the |0) state should be excited with probability p11.
# gamma must be scaled to reflect this.
gamma = p11 / p = p00 / (1 - p)
```